pkg/decision/forced_decision_service.go

```diff
@@ -0,0 +1,124 @@
+/****************************************************************************
+ * Copyright 2021, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package decision //
+package decision
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/optimizely/go-sdk/pkg/config"
+	"github.com/optimizely/go-sdk/pkg/decide"
+	"github.com/optimizely/go-sdk/pkg/entities"
+)
+
+type forcedDecision struct {
+	flagKey string
+	ruleKey string
+}
+
+// ForcedDecisionService defines user contexts that the SDK will use to make decisions for.
+type ForcedDecisionService struct {
+	UserID          string
+	forcedDecisions map[forcedDecision]string
```

Comment:
`Structs` can be used as keys to a map if they hold comparable values which in our case is true since we have string values inside `forcedDecision `. https://go.dev/blog/maps 
I have benchmarked both map of maps and the above approach. Both approaches take almost equal time to read and write.

Reply:
@msohailhussain Since go allows for the use of structs as keys then I think this is fine. My assumption is that the struct becomes some hash value in the background and that is what allows for quick lookups, same as if it were a single value. The only thing we would need to be careful of is if later we extend the forced decision struct to contain more data, as if we then update that object, the key will be different and take more than one place in the map if we are not careful and removing the older instance. This can also lead to a memory leak possibly if we dont clean up old objects that "get lost in the map". I am not 100% sure the Go implementation of this, but in Java that is an issue and why it should be used very cautiously. @jaeopt Do we see any extension of the forcedDecisions in the future? If so I would say avoid this method, otherwise - it should be alright for now.

Reply:
agree !!!

Reply:
OK I checked it and struct is much faster than conventional mapOfMap.
```
2021/09/29 11:50:43 mapOfMaps took 545.791µs
2021/09/29 11:50:43 mapOfForcedDecisions took 270.387µs
2021/09/29 11:50:43 mapOfMaps took 460.888µs
2021/09/29 11:50:43 mapOfForcedDecisions took 294.126µs
2021/09/29 11:50:43 mapOfMaps took 498.434µs
2021/09/29 11:50:43 mapOfForcedDecisions took 240.871µs
```

Reply:
Insertion is bit slower in struct keys. 
```
2021/09/29 12:20:50 mapOfMaps took 914.17801ms
2021/09/29 12:20:53 forcedDecision took 2.856786962s
```

Reply:
It's not likely to extend FD in future more than what we have, but cannot be guaranteed. I think it'll be safe as long as we have good test cases. Performance-wise, I don't care much since this is for testing environment only :) @msohailhussain @The-inside-man